CmsManager/CmsManagerSelector.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DashboardBundle\CmsManager;
+
+use Sonata\DashboardBundle\Exception\DashboardNotFoundException;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\SecurityContextInterface;
+use Symfony\Component\Security\Http\Event\InteractiveLoginEvent;
+use Symfony\Component\Security\Http\Logout\LogoutHandlerInterface;
+
+/**
+ * This class return the correct manager instance :
+ *   - sonata.dashboard.cms.dashboard if the user is an editor (ROLE_SONATA_DASHBOARD_ADMIN_PAGE_EDIT)
+ *   - not found exception if the user is a standard user.
+ *
+ * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ */
+final class CmsManagerSelector implements CmsManagerSelectorInterface, LogoutHandlerInterface
+{
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    /**
+     * @param ContainerInterface $container
+     */
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function retrieve()
+    {
+        if ($this->isEditor()) {
+            $manager = $this->container->get('sonata.dashboard.cms.dashboard');
+        } else {
+            // No manager implemented yet if the user is a standard user
+            throw new DashboardNotFoundException('Unable to retrieve the cms manager');
+        }
+
+        return $manager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isEditor()
```

Comment:
@core23 "because the functionality is really basic": the dashboard/block manager has to be developed because nothing exists by now, or you already have something ?